### ISSUE TITLE
[question] Decreasing ep_rew_mean (reward function maximizes utilization rate) while loss also converges to 0

### ISSUE THREAD

**Issue by 16mkor** [DLR-RM/stable-baselines3#1375]
### 🐛 Bug

I am training an RL agent to optimise dispatching in a job shop manufacturing system. 
## Tensorboard Diagrams
During training I noticed that **the mean reward per episode, the ep_re_mean in my tensorboard, decreases over time contrary to my expectation** that it should be increasing. The reward function is the utilization rate of the machines and should be maximised, which is why it is positive and calculated as `r_util = exp(util/1.5) - 1`. I am using a custom gym environment and a simpy environment. 

![ep_rew_mean](https://user-images.githubusercontent.com/104545181/224507976-fef94c46-871f-41b2-8ccc-917945b6fd09.PNG)

The loss functions, however, converge to 0 (except for the DQN in blue). 

![loss_!](https://user-images.githubusercontent.com/104545181/224507977-1d4f42cf-d3c9-4dfb-a898-0e8b271211f6.PNG)
![loss_2](https://user-images.githubusercontent.com/104545181/224507978-6b954c09-29fa-447b-91a6-4d2f5a56535f.PNG)

What could be the reason for this behaviour?

## What I tried so far.

I already tried to check on the reward function, but as far as I understand, it works how I expect it to work.. Also, I checked if the reward transferred to the tensorboard is similar to the reward in my logging files. I read the post here [Why does ep_re_mean decrease over time?](https://stackoverflow.com/questions/72105659/why-does-ep-rew-mean-decrease-over-time), but it did not help..
Does anyone have any idea why the mean episodic reward decreases over time?

### Code example

```python
import gym
import numpy as np
from gym import spaces

from stable_baselines3 import A2C
from stable_baselines3.common.env_checker import check_env


class CustomEnv(gym.Env):

    def __init__(self):
        super().__init__()
        self.observation_space = spaces.Box(low=-np.inf, high=np.inf, shape=(14,))
        self.action_space = spaces.Box(low=-1, high=1, shape=(6,))

    def reset(self):
        return self.observation_space.sample()

    def step(self, action):
        obs = self.observation_space.sample()
        reward = 1.0
        done = False
        info = {}
        return obs, reward, done, info

env = CustomEnv()
check_env(env)

model = A2C("MlpPolicy", env, verbose=1).learn(1000)
```

### Relevant log output / Error message

```shell
Agent Reward Log:

episode,sim_step,sim_time,action,reward,action_valid,state
0.0,0.0,0.0,[8],-0.0,False,0.0
1.0,1.0,0.0,[3],-0.0,True,"[ 1.00000000e+00  1.00000000e+00  0.00000000e+00  1.00000000e+00
  1.00000000e+00  0.00000000e+00  1.00000000e+00  1.00000000e+00
  0.00000000e+00  0.00000000e+00  1.00000000e+00  1.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
  1.00000000e+00  1.00000000e+00  1.00000000e+00  1.00000000e+00
  1.00000000e+00  1.00000000e+00  1.00000000e+00  1.00000000e+00
  1.00000000e+00  1.00000000e+00  1.00000000e+00  1.00000000e+00
  1.00000000e+00  1.00000000e+00  1.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  1.00000000e+00
  1.00000000e+00  1.00000000e+00  0.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
  1.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
  2.00000000e+00  2.00000000e+00  2.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  6.66666667e-01
  6.66666667e-01  6.66666667e-01  1.00000000e+00  1.00000000e+00
  1.00000000e+00  1.00000000e+00  1.00000000e+00  1.00000000e+00
  1.00000000e+00  1.00000000e+00  1.00000000e+00  1.00000000e+00
  1.00000000e+00  1.00000000e+00  1.00000000e+00  1.00000000e+00
  1.00000000e+00  1.00000000e+00  3.33333333e-01  3.33333333e-01
  3.33333333e-01  0.00000000e+00  0.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
  0.00000000e+00  1.00000000e-06  1.00000000e-06  1.00000000e-06
 -1.00000000e+00 -1.00000000e+00 -1.00000000e+00 -1.00000000e+00
 -1.00000000e+00 -1.00000000e+00 -1.00000000e+00 -1.00000000e+00
  1.00000000e+00  1.00000000e+00  0.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00]"
1.0,2.0,2.66667,[21],-0.0,True,"[ 1.          1.          0.          1.          1.          1.
  1.          1.          1.          0.          1.          1.
  0.          0.          0.          0.          0.          0.
  0.          0.          1.          1.          1.          1.
  1.          1.          1.          1.          1.          1.
  1.          1.          1.          1.          1.          0.
  0.          0.          0.          0.          0.          0.
  0.          1.          1.          1.          0.          0.
  0.          1.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          3.          3.
  3.          0.          0.          0.          0.          0.
  0.          0.          0.          1.          1.          1.
  1.          1.          1.          1.          1.          1.
  1.          1.          1.          1.          1.          1.
  1.          1.          1.          1.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          7.16666567  8.          8.
 -1.         -1.         -1.         -1.         -1.         -1.
 -1.         -1.          1.          1.          1.          0.
  0.          0.         78.34123014  0.          0.          0.
  0.          0.          0.          0.         78.34123014  0.
  0.          0.          0.        ]"
1.0,3.0,3.5,[24],-0.0,True,"[ 1.          1.          0.          1.          1.          1.
  1.          1.          1.          0.          1.          1.
  0.          0.          0.          0.          0.          0.
  0.          0.          1.          1.          1.          1.
  1.          1.          1.          1.          1.          1.
  1.          1.          1.          1.          1.          0.
  0.          0.          0.          0.          0.          0.
  0.          1.          1.          1.          0.          1.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          3.          3.
  3.          0.          0.          0.          0.          0.
  0.          0.          0.          1.          1.          1.
  1.          1.          1.          1.          1.          1.
  1.          1.          1.          1.          1.          1.
  1.          1.          1.          1.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          9.66666567 10.5        10.5
 -1.         -1.         -1.         -1.         -1.         -1.
 -1.         -1.          1.          1.          1.          0.
  0.          0.         77.5078968   0.          0.          0.
  0.          0.          0.          0.         77.5078968   0.
  0.          0.          0.        ]"
1.0,4.0,4.33334,[26],-0.0,True,"[ 1.          1.          0.          1.          1.          1.
  1.          1.          1.          0.          1.          1.
  0.          0.          0.          0.          0.          0.
  0.          0.          1.          1.          1.          1.
  1.          1.          1.          1.          1.          1.
  1.          1.          1.          1.          1.          0.
  0.          0.          0.          0.          0.          0.
  0.          1.          1.          1.          0.          0.
  0.          0.          1.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          3.          3.
  3.          0.          0.          0.          0.          0.
  0.          0.          0.          1.          1.          1.
  1.          1.          1.          1.          1.          1.
  1.          1.          1.          1.          1.          1.
  1.          1.          1.          1.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.         12.16666567 13.         13.
 -1.         -1.         -1.         -1.         -1.         -1.
 -1.         -1.          1.          1.          1.          0.
  0.          0.         76.67456347  0.          0.          0.
  0.          0.          0.          0.         76.67456347  0.
  0.          0.          0.        ]"
1.0,5.0,5.16667,[4],-0.0869,True,"[ 1.          1.          0.          1.          1.          1.
  1.          1.          1.          0.          1.          1.
  0.          0.          0.          0.          0.          0.
  0.          0.          1.          1.          1.          1.
  1.          1.          1.          1.          1.          1.
  1.          1.          1.          1.          1.          0.
  0.          0.          0.          0.          0.          0.
  0.          1.          1.          1.          0.          0.
  0.          0.          0.          0.          1.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          3.          3.
  3.          0.          0.          0.          0.          0.
  0.          0.          0.          1.          1.          1.
  1.          1.          1.          1.          1.          1.
  1.          1.          1.          1.          1.          1.
  1.          1.          1.          1.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.         14.66666567 15.5        15.5
 -1.         -1.         -1.         -1.         -1.         -1.
 -1.         -1.          1.          1.          1.          0.
  0.          0.         75.84123014  0.          0.          0.
  0.          0.          0.          0.         75.84123014  0.
  0.          0.          0.        ]"
```

### System Info

Technical Details:
- Using Python 3.10 and pip3
- Relevant requirements:
- gym==0.21.0
- gymnasium @ git+https://github.com/pseudo-rnd-thoughts/Gymnasium@af553a25533168aacf1dec40ffac03663146a31d
- huggingface-sb3==2.2.4
- keras==2.11.0
- simpy==4.0.1
- tensorboard==2.11.2
- tensorflow==2.11.0
- torch==1.13.1

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] I have provided a minimal working example to reproduce the bug
- [X] I have checked my env using the env checker
- [X] I've used the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by qgallouedec** [comment]
Have you tried using another algorithm? Additionally, it may be worth checking your custom environment implementation. You could start with a very basic environment and gradually increase the complexity to see where the problem occurs. However, please keep in mind that as a maintainer, we are not able to provide technical support for individual issues. I recommend joining the RL Discord server, where you may be able to find someone who can offer more assistance.

**Comment by 16mkor** [comment]
Thank you for your answer! I reached out to the Discord Server for more help. Also, I will try to rebuild my environment to see if I can find the causing problem. 
If someone else has an idea I am still happy for other thoughts:)
